src/client/Microsoft.Identity.Client/ApiConfig/GetAuthorizationRequestUrlParameterBuilder.cs

```diff
@@ -97,6 +100,32 @@ public GetAuthorizationRequestUrlParameterBuilder WithPkce(out string codeVerifi
             return this;
         }
 
+        /// <summary>
+        /// To help with resiliency, AAD Cached Credential Service (CCS) operates as an AAD backup.
+        /// </summary>
+        /// <param name="userObjectIdentifier">GUID which is unique to the user, parsed from the client_info.</param>
+        /// <param name="tenantIdentifier">GUID format of the tenant ID, parsed from the client_info.</param>
+        /// <returns>The builder to chain the .With methods</returns>
+        public GetAuthorizationRequestUrlParameterBuilder WithCcsRoutingHint(string userObjectIdentifier, string tenantIdentifier)
```

Comment:
GetAuthorizationUrl doesn't make any network calls, it just creates an URL.

Reply:
but it creates the URL for the challenge of the user, Bogdan, which will be used in incremental consent, or claims challenge in ASP.NET (not core) applications. It needs to have the ccs routing attributes

Reply:
The URL will have the CCS info as a GET params (query params), because this API returns an Uri, not an HttpRequestMessage. So Asp.NET or any other consumers would have to extract the GET params and move them to a header. Do you expect this to happen?